temporalio/worker/_workflow.py

```diff
@@ -250,9 +316,32 @@ async def _handle_activation(
                         activate_task, self._deadlock_timeout_seconds
                     )
                 except asyncio.TimeoutError:
-                    raise RuntimeError(
-                        f"[TMPRL1101] Potential deadlock detected, workflow didn't yield within {self._deadlock_timeout_seconds} second(s)"
-                    )
+                    # extract the stack of the workflow if possible
+
+                    # cases:
+                    # 1. get_thread_id() is not implemented for the worker (returns None, fall to case 2)
+                    # 2. get_thread_id() returns None
+                    # 3. tid is not present in sys._current_frames()
+                    # 4. error occurs during traceback construction
+                    # 5. normal
+
+                    msg = f"[TMPRL1101] Potential deadlock detected: workflow didn't yield within {self._deadlock_timeout_seconds} second(s)."
+                    tid = workflow.get_thread_id()
```

Comment:
I think it's important that we swallow any exceptions that could occur while building this. It's extra nice-to-have data, not required and should not affect showing this exception.